### ISSUE TITLE
The way permeability works is atrocious, permeability checks are rolled INDIVIDUALLY FOR EACH REAGENT IN THE REACTION

### ISSUE THREAD

**Issue by 9600bauds** [d3athrow/vgstation13#7786]
So here's how it works.
Someone splashes you with a beaker (or sprays you with a hand sprayer, or uses grenade smoke on you, or puts you in cryo, etc). If the chemicals that were in the beaker have any on_touch special effects (polyacid, mutagen, and maybe 2 more things) they will do their own thing. Afterwards, the chemicals in the beaker have a chance to be added directly into your bloodstream. 
Example 1: Someone splashes you with a beaker of polyacid. Polyacid has a very strong on_touch effect which basically fucks you up or melts your mask/helmet. After the special on_touch effect, there is a chance for the polyacid to go directly into your bloodstream.
Example 2: Someone splashes you with a beaker of chloral, but since the chloral doesn't have any special on_touch effects, the only thing that will happen is the chance for chloral to go directly into your bloodstream.

Sounds good? Sounds simple?

Now, the chance to block the puff of chemicals depends on what clothing you're wearing. Specifically, it goes through every single clothing item that you have equipped, and gets the lowest permeability_coefficient out of them all. That, multiplied by 100, is the percentage chance that the chemicals have of affecting you
Did that make sense? Let's think about it. Your shoes have a permeability_coefficient of 0.5, which means wearing shoes gives a beaker of chloral a 50% of working when splashed on you. Okay, I guess the code thinks that shoes cover your entire body or something.
Nearly all gloves have a permeability_coefficient of 0.05 or under. That means that JUST wearing gloves makes chloral to the face have less than a 5% chance of working, even if you aren't wearing anything except the gloves.

Okay, wow, that's pretty stupid.
But wait. This chance for the chemicals to go through IS ROLLED INDIVIDUALLY FOR EACH CHEMICAL IN THE BEAKER.
So if you splash someone with a beaker that has both chloral AND cyanide, then you have a twice as large chance to knock the guy out (which is still individually 1-5% depending on which gloves or mask he has).

Wow. That's absolutely terrible.

It's worth noting that Cryo "dupes" reagents by beakersplashing you magically. So, IF YOU PUT AN ASSISTANT WITH A GAS MASK INTO CRYO YOU'RE ESSENTIALLY SUCKING UP THE CRYOMIX 99 TIMES AS FAST. Uh-oh, wait, not just that. You also have a chance for every individual chemical in the cryomix to actually go through. That means that you could be unlucky enough to have Cryoxadone go through, but not Clonexadone, thus effectively putting the assistant on a roundstart cryomix of just 100% Cryox, all fucking randomly.

**Comment by ihadtoregisterforthis** [comment]
Is this oldcode, shitcode or specific defunct coder code?

**Comment by 9600bauds** [comment]
I'm fairly sure it's all

**Comment by 9600bauds** [comment]
Also, and this is especially hilarious to me, the bio-hood and bio-suit have a special hardcode exception.

**Comment by Exxion** [comment]
It should probably use the highest permeability coefficient of exposed clothes (as in ones that show up on examine) or zero if any body part is uncovered, if it has to use full-body checks

Er... does a space suit cover your shoes?

**Comment by Shadowmech88** [comment]
Hardsuits and space suits both cover your feet and your hands. Sprite-wise and common-sense-wise anyway.
Not sure if the code counts them as covered.

**Comment by ghost** [comment]
When Chemistry-Reagents.dm was reworked, the TOUCH and INGEST differences were kept as is

At least now it's cleaned up somewhat

**Comment by 9600bauds** [comment]
Also, the Revialver bypasses this entirely. The shots always go through. They don't account for permeability at all.

**Comment by Exxion** [comment]
To be honest I'm not sure that's worse

**Comment by 9600bauds** [comment]
Shadowmech said he wanted it to work differently from a syringe gun, but that's just not doable with the current permeability code
Shame

**Comment by Shadowmech88** [comment]
I coded it so that a revialver impact should use the same code that regular splashing does.

Does it behave differently than regular splashing?

**Comment by 9600bauds** [comment]
Unless I got unlucky (or lucky?) enough to have a 1% chance happen 3 times in a row, then yes, somehow
Normally a beaker splash has 99% to fail when wearing a gas mask, but that didnt stop the revialver

**Comment by Shadowmech88** [comment]
Weird.

**Comment by Kurfursten** [comment]
The fact that this hasn't been refactored 5 months later is a testament to how much nobody wants to touch permeability. Thanks for at least doing the research bauds.

**Comment by PJB3005** [comment]
No, it's a testament to how little people give a shit about the issue tracker in general.

**Comment by ComicIronic** [comment]
We have an issue tracker? I thought this was just a PR graveyard.

**Comment by ihadtoregisterforthis** [comment]
It's actually a feature request blog
